### PR TITLE
multi: bump lnd version to v0.17.0-beta

### PR DESCRIPTION
**Pull request by guggero**
Prepares everything to update to `lnd v0.17.0-beta` once the final release is out (we'll want to wait with merging this until then).

#### Pull Request Checklist
- [ ] Update `release_notes.md` if your PR contains major features, breaking changes or bugfixes

### CODE REVIEW

File: loopdb/sqlite.go [hunk 1]
```diff
@@ -211,14 +211,19 @@ func (db *BaseDB) ExecTx(ctx context.Context, txOptions TxOptions,
...
 func (b *BaseDB) FixFaultyTimestamps(ctx context.Context) error {
-
 	// Manually fetch all the loop out swaps.
 	rows, err := b.DB.QueryContext(
 		ctx, "SELECT swap_hash, publication_deadline FROM loopout_swaps",
 	)
 	if err != nil {
 		return err
 	}
+	if rows.Err() != nil {
```
Comment (bhandras) [inline loopdb/sqlite.go:221]: Nice catch! I believe instead what we'd want is first go through the rows (line 237), then after that `for` loop we'd:
```
	if err := rows.Close(); err != nil {
		return nil, err
	}
	if err := rows.Err(); err != nil {
		return nil, err
	}
```

**Review by bhandras — APPROVED** [review]
Nice fixes, thanks! 🎉

File: loopdb/sqlite.go (hunk 1 above)
Comment (guggero) [reply loopdb/sqlite.go]: Ah, good point. Though the linter really wants me to call `Close` with a `defer`... So how does that latest version look to you?

**Review by bhandras — APPROVED** [review]

File: loopdb/sqlite.go (hunk 1 above)
Comment (bhandras) [reply loopdb/sqlite.go]: Looks good!

**Review by ellemouton — COMMENTED** [review]
lgtm once updated to rc6 ⚡

**Comment by guggero** [comment]
Bumped to final version `lnd v0.17.0-beta`, will merge after green CI.

**Pull request merged**